docs/rfcs/blocks-in-widget-areas.md

```diff
@@ -0,0 +1,520 @@
+- Body: A list of widget area resources
+
+```json
+[
+	{
+		"id": "footer",
+		"name": "Footer",
+		"description": "",
+		"content": {
+			"raw": "<!-- wp:paragraph --><p>Hello there!</p><!-- /wp:paragraph -->",
+			"rendered": "<p>Hello there!</p>"
+		},
+	},
+	{
+		"id": "sidebar",
+		"name": "Sidebar",
+		"description": "",
+		"content": {
+			"raw": "<!-- wp:paragraph --><p>Hello there!</p><!-- /wp:paragraph -->",
+			"rendered": "<p>Hello there!</p>"
+		},
+	}
+]
+```
+
+### Fetching a single widget area
+
+Fetches a single registered widget area by its ID.
+
+#### Request
+
+- Method: `GET`
+- Route: `/wp/v2/widget-areas/:id`
+- Parameters:
+  - `id`: The unique identifier of the widget area to fetch
+
+```
+GET /wp/v2/widget-areas/footer
+```
+
+#### Response
+
+- Status:
+  - `200` if successful
+  - `404` if requested widget area does not exist
+- Body: A single widget area resource
+
+```json
+{
+	"id": "footer",
+	"name": "Footer",
+	"description": "",
+	"content": {
+		"raw": "<!-- wp:paragraph --><p>Hello there!</p><!-- /wp:paragraph -->",
+		"rendered": "<p>Hello there!</p>"
+	},
+}
+```
+
+### Updating a single widget area
+
+Updates the blocks in a single registered widget area.
+
+#### Request
+
+- Method: `PUT`
+- Route: `/wp/v2/widget-areas/:id`
+- Parameters:
+  - `id`: The unique identifier of the widget area to update
+- Body: A single widget area resource
+
+```
+PUT /wp/v2/widget-areas/footer
+
+{ "content": { "raw": "<!-- wp:block {\"ref\":123} /-->" } }
+```
+
+#### Response
+
+- Status:
+  - `200` if successful
+  - `404` if specified widget area does not exist
+  - `400` if request body is not a valid widget area resource
+- Body: The updated widget area resource
+
+```json5
+{
+	"id": "footer",
+	"name": "Footer",
+	"description": "",
+	"content": {
+		"raw": "<!-- wp:block {\"ref\":123} /-->",
+		"rendered": "<p>This is a reusable paragraph block</p>"
+	},
+}
+```
+
+## Storage
+
+WordPress currently stores which widgets belong to which widget areas using the `'sidebars_widgets'` site option. This is a serialized PHP array that maps widget area identifiers to a list of widget identifiers that are in that area.
+
+```php
+array(
+	'footer' => array(
+		'search-2',
+		'recent-posts-2',
+		'recent-comments-2',
+	),
+	'sidebar' => array(
+		'recent-posts-2',
+		'recent-comments-2',
+	),
+	'array_version' => 3,
+)
+```
+
+When a widget area stored this way is accessed via the REST API, widgets will be represented using `core/legacy-widget` blocks. Each `core/legacy-widget` block stores a reference to the existing widget (`identifier`), and, if it is a multi widget, the attributes that that instance has (`instance`).
+
+```
+GET /wp/v2/widget-areas/footer
+```
+
+```json
+{
+	"id": "footer",
+	...
+	"content": {
+		"raw": "
+			<!-- wp:legacy-widget {\"identifier\":\"search-2\",\"instance\":{\"title\":\"\"}} /-->
+			<!-- wp:legacy-widget {\"identifier\":\"recent-posts-2\",\"instance\":{\"title\":\"\",\"number\":5}} /-->
```

Comment:
This identifier is referencing existing widget instances. In legacy widget block we are using classes as the identifier (for class widgets) to avoid the need for instance creation.  Should we support class identifiers? Do you see any cons in that?
If not supporting class identifiers? Does the endpoint support updating existing widgets? Does the endpoint support creating new widget instances how that creation happens?